department-of-veterans-affairs/caseflow
# added stmdtime update to update_veteran_sfnod

Resolves [APPEALS-13049](https://vajira.max.gov/browse/APPEALS-13049)

### Description
- Added STMDTIME update to update_veteran_sfnod method for MPI person updates
- Added script update_vacols_stmdtime_from_caseflow.rb to retroactively update CORRES.STMDTIME with the MpiUpdatePersonEvent.completed_at value for any records that have not been updated since the SFNOD was updated from Caseflow
- Added script add_corres_for_stmdtime_testing.rb to generate data locally in both VACOLS and Caseflow for testing

### Acceptance Criteria
- [ ] Script runs without error

### Testing Plan
For controller testing in UAT:
1. Go into Metabase and view FACOLS (appeals-dev-vacols) and find a veteran
2. View the corres table
3. Filter by the veteran's PAT or SSN in step 1
4. Use that veteran and do https://github.com/department-of-veterans-affairs/appeals-lambdas/tree/master/mpi-soap#testing
5. Refresh the Metabase view
6. Observe stmdtime is updated for the time the request was sent

For local testing with scripts:
1. Run the add_corres_for_stmdtime_testing.rb script to generate data
2. Run the update_vacols_stmdtime_from_caseflow.rb
3. Verify that the desired updates have been made (see comments at top of data script)
